aequilibrae/project/network/network.py

```diff
@@ -0,0 +1,198 @@
+                raise ValueError("Coordinates out of bounds")
+            bbox = [west, south, east, north]
+        else:
+            bbox, report = placegetter(place_name)
+            west, south, east, north = bbox
+            if bbox is None:
+                msg = 'We could not find a reference for place name "{}"'.format(place_name)
+                warn(msg)
+                logger.warn(msg)
+                return
+            for i in report:
+                if "PLACE FOUND" in i:
+                    logger.info(i)
+
+        # Need to compute the size of the bounding box to not exceed it too much
+        height = haversine((east + west) / 2, south, (east + west) / 2, north)
+        width = haversine(east, (north + south) / 2, west, (north + south) / 2)
+        area = height * width
+
+        if area < max_query_area_size:
+            polygons = [bbox]
+        else:
+            polygons = []
+            parts = math.ceil(area / max_query_area_size)
+            horizontal = math.ceil(math.sqrt(parts))
+            vertical = math.ceil(parts / horizontal)
+            dx = east - west
+            dy = north - south
+            for i in range(horizontal):
+                xmin = max(-180, west + i * dx)
+                xmax = min(180, west + (i + 1) * dx)
+                for j in range(vertical):
+                    ymin = max(-90, south + j * dy)
+                    ymax = min(90, south + (j + 1) * dy)
+                    box = [xmin, ymin, xmax, ymax]
+                    polygons.append(box)
+
+        logger.info("Downloading data")
+        self.downloader = OSMDownloader(polygons, modes)
+        self.downloader.doWork()
+
+        logger.info("Building Network")
+        self.builder = OSMBuilder(self.downloader.json, self.conn)
+        self.builder.doWork()
+
+        if spatial_index:
+            logger.info("Adding spatial indices")
+            self.add_spatial_index()
+
+        self.add_network_triggers()
+        logger.info("Network built successfully")
+
+    def create_empty_tables(self) -> None:
+        curr = self.conn.cursor()
+        # Create the links table
+        p = Parameters()
+        fields = p.parameters["network"]["links"]["fields"]
+
+        sql = """CREATE TABLE 'links' (
+                          ogc_fid INTEGER PRIMARY KEY,
+                          link_id INTEGER UNIQUE NOT NULL,
+                          {});"""
+
+        flds = fields["one-way"]
+
+        owlf = [
+            "{} {}".format(list(f.keys())[0], f[list(f.keys())[0]]["type"])
+            for f in flds
+            if list(f.keys())[0].upper() != "LINK_ID"
+        ]
+
+        flds = fields["two-way"]
+        twlf = []
+        for f in flds:
+            nm = list(f.keys())[0]
+            tp = f[nm]["type"]
+            twlf.extend(["{}_ab {}".format(nm, tp), "{}_ba {}".format(nm, tp)])
```

Comment:
Common py3.6+

https://realpython.com/python-f-strings/

Reply:
Wow!! I didn't know this one!!

Reply:
https://media0.giphy.com/media/xT0xeJpnrWC4XWblEk/giphy.gif

Reply:
I also didn't know we could use GIFs here.... Hahahaha

Reply:
I went ahead and changed it in a lot of places.  Might have a future PR just to change that

Reply:
just note that f-strings are 3.6+